lib/myxql/protocol/values.ex

```diff
@@ -129,7 +129,10 @@ defmodule MyXQL.Protocol.Values do
   end
 
   def decode_text_value(value, type) when type in [:float, :double] do
-    String.to_float(value)
+    case Float.parse(value) do
```

Comment:
Performance wise checking for a dot and using String.to_float or String.to_integer should be much faster.

Reply:
@greg-rychlewski, any object if i switch back to...

```elixir
value = if String.contains?(value, "."), do: value, else: "#{value}.0"
String.to_float(value)
```

We can probably drop the raise then, too, because it's very unlikely to fail.

Reply:
> my bad, sorry @benkimpel

all good!

Reply:
Updated PR coming up

Reply:
I believe Jose was thinking something like this

```elixir
if String.contains?(value, ".") do
  String.to_float(value)
else
  String.to_integer(value)
end
```
Then we can save one iteration over the original string here: `"#{value}.0"`

Reply:
I think we still want a float out at the end, though.

Reply:
I think since we are at the driver level and not Ecto, we probably want to represent what is given by MySQL as faithfully as possible rater than coercing types. But I see your point and this is probably not a completely straight forward decision. Let's get Jose's opinion as well.

Reply:
> But I see your point and this is probably not a completely straight forward decision. Let's get Jose's opinion as well.

Sounds good.

@josevalim let us know what you think.

Reply:
For some more context around why I've suddenly appeared with issues about a TextQuery...

We're using an OLAP database called StarRocks which uses the MySQL protocol. (I believe there are a few other non-MySQL DBs aside from MariaDB out there that do the same. It's a nice protocol.) 

We don't want to have to do it, but there is a scenario where we might need to use this driver directly. In that situation getting a float would be best/most accurate, because `Ecto` might not even be in the picture.

---

A much bigger discussion... :wink:

There's a few other things that might prevent us from using this specific driver for StarRocks:

1. StarRocks supports prepared statements, but they do not currently support it in their MySQL protocol implementation. (I've been in contact with the devs and have opened an issue. Not sure on roadmap/prioritization.)
2. `Ecto` seems to always use prepared statements.
3. `MyXQL` requires some flags that StarRocks (and probably many OLAP DBs) don't support: `client_found_rows`, `client_multi_results`, `client_multi_statements`, and `client_transactions`.

We'll probably fork `MyXQL` and the `Ecto.Adapters.MyXQL` to see if we can get this all running. We're not sure if it could ever merge back into `MyXQL`, but in order to do so we'd need to be able to set those client flags as optional and execute text queries.

Reply:
It makes sense. The one point I was trying to make though is that I don't think it's a driver issue but rather an issue with how MySQL returns those values in their text protocol. For instance, when I use the official mysql client on my laptop I get this

```sql
mysql> create table float_test (x float);
Query OK, 0 rows affected (0.03 sec)

mysql> insert into float_test(x) values (10.0);
Query OK, 1 row affected (0.00 sec)

mysql> select * from float_test;
+------+
| x    |
+------+
|   10 |
+------+
1 row in set (0.00 sec)
```

Reply:
Ah, i see where you're coming from. Gotcha.

Reply:
@greg-rychlewski for me the fact it displays 10.0 as "10" is a representation decision, but it is known that the type  is float. We should then parse it to float, independent from the representation.

Reply:
That makes sense. The only thing though is that this is the representation chosen by the protocol we are following rather than a specific client.

Reply:
Maybe it is better to parse it as float. If we kept everything as text and left it up to the user to convert it then keeping "10" would be best. But now we are trying to do the best conversions for them. So in this case it would be float.

Reply:
> That makes sense. The only thing though is that this is the representation chosen by the protocol we are following rather than a specific client.

FWIW, I took a look at the official MySQL python driver and it returns floats in this case. Not sure about the others or if it is safe to make assumptions based on the behavior in Python... but it is official.

I'll admit I haven't been able to find a good reference for this representation in the protocol. Thinking about why it might be the way it is, it's conceivable that they're trying to minimize packet size as much as possible.